models/progresses.js

```diff
@@ -47,9 +48,40 @@ const createProgressDocument = async (progressData) => {
  * @throws {Error} If the userId or taskId is invalid or does not exist.
  **/
 const getProgressDocument = async (queryParams) => {
+  const { dev } = queryParams;
   await assertUserOrTaskExists(queryParams);
   const query = buildQueryToFetchDocs(queryParams);
   const progressDocs = await getProgressDocs(query);
+
+  if (dev) {
+    try {
+      const uniqueUserIds = [...new Set(progressDocs.map((doc) => doc.userId))];
+      const batchSize = 500;
+
+      const batches = Array.from({ length: Math.ceil(uniqueUserIds.length / batchSize) }, (_, index) =>
+        uniqueUserIds.slice(index * batchSize, index * batchSize + batchSize)
+      );
+
+      const batchPromises = batches.map((batch) => Promise.all(batch.map((userId) => fetchUser({ userId }))));
+
+      const usersDataBatches = await Promise.all(batchPromises);
+
+      const usersData = usersDataBatches.flat();
+
+      const userLookupMap = usersData.reduce((lookup, { user }) => {
+        if (user) lookup[user.id] = user;
```

Comment:
write tests for this.

Reply:
Hello @Achintya-Chatterjee I just need some clarification regarding the tests. I can add a test to check that userData returns null if the operation fails, but I'm also wondering how to test for scenarios with a large number of unique users. With our current test setup, I would need to add 1000+ users to Firestore to test it.

If you want me to proceed with that approach, please let me know!